File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n) {
    vector<int> answer;
    
    answer.push_back(n);
    
    while(n != 1) {
        (n % 2 == 0) ? (n = n / 2) : (n = 3 * n + 1);
        answer.push_back(n);
    }
    
    return answer;
}


/*
    짝수일 때 2로 나누고.
    홀수일 때 3을 곱하고 1을 더한다.
    그 결과 수열을 반환하라.
*/

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.swift
import Foundation

func solution(_ n:Int) -> [Int] {
    var result: [Int] = []
    var x = n
    
    while x != 1 {
        result.append(x)
        x.isMultiple(of: 2) ? (x /= 2) : (x = x * 3 + 1)
    }
    
    result.append(1)
    
    return result
}
